### PR TITLE
Fix scroll behaviour on skill example, chatApp code

### LINKED ISSUE #2673: Fix redux code for chatApp, testExample
**Actual Behaviour**

Current code is not efficiently written

**Expected Behaviour**

Refactor and fix the code.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by fragm3**
Fixes #2673, #2665

Changes: 
- [x] Fix scroll behavior on skill example
- [x] Fix chatApp code for the chat bubble
- [x] Created a generic generateMessage function
- [x] Fix page scroll on active search messages found
- [x] Remove blue, black outline on active and hover of Search

Screenshot 2019-07-27 at 10 24 42 PM

Demo Link: https://pr-2674-fossasia-susi-web-chat.surge.sh

### CODE REVIEW

**Comment by fragm3** [comment]
@harsh-jindal fixed it

**Comment by harsh-jindal** [comment]
@fragm3 The issue with scrollbar still exists. If you have a couple messages and open up the search bar and try to scroll to the top, as soon as the scrollbar reaches the top, the chat scrolls to the bottom in an abrupt manner.
Issue with question not showing up for the first time on test skill example is resolved.

**Comment by fragm3** [comment]
Working fine for me @harsh-jindal, fixed it.

Jul-27-2019 21-41-24

**Comment by harsh-jindal** [comment]
@fragm3 Try it without writing anything in the search field.
Steps:
open the chatbot.
click on search icon. (don't write or focus on the search field)
scroll to the top.

**Comment by fragm3** [comment]
@harsh-jindal fixed

File: src/components/ChatApp/MessageComposer.react.js [hunk 1]
```diff
@@ -369,25 +351,15 @@ class MessageComposer extends Component {
 
   createUserMessage(text, voice) {
     this.userMessageHistory = [text, ...this.userMessageHistory];
-    formatUserMessage({
+    console.log(this.userMessageHistory, 'userMessageHistory');
```
Comment (harsh-jindal) [inline src/components/ChatApp/MessageComposer.react.js]: Remove this.

**Comment by fragm3** [comment]
@harsh-jindal made the changes

File: src/App.js [hunk 2]
```diff
@@ -187,11 +188,10 @@ class App extends Component {
             )}
             {renderAppBar}
             {renderAppBanner}
-            {pathname !== '/chat' ? <ChatApp showChatBubble={true} /> : null}
+            {pathname !== '/chat' ? <ChatApp /> : null}
```
Comment (harsh-jindal) [inline src/App.js]: This can be just `<ChatApp />`

**Comment by fragm3** [comment]
@harsh-jindal done

File: src/components/ChatApp/MessageComposer.react.js [hunk 3]
```diff
@@ -235,33 +229,21 @@ class MessageComposer extends Component {
   }
 
   componentDidMount() {
-    const { actions, testSkillExampleQuery } = this.props;
+    const { actions, pendingUserMessage, speechOutputAlways } = this.props;
+    console.log({ pendingUserMessage });
```
Comment (akshatnitd) [inline src/components/ChatApp/MessageComposer.react.js]: Remove

File: src/components/ChatApp/MessageComposer.react.js [hunk 4]
```diff
@@ -527,7 +491,13 @@ function mapDispatchToProps(dispatch) {
...
 
 function mapStateToProps(store) {
-  const { messagesByID, messages, loadingHistory } = store.messages;
+  const {
+    messagesByID,
+    messages,
+    loadingHistory,
+    pendingUserMessage,
+  } = store.messages;
+  console.log({ pendingUserMessage });
```
Comment (akshatnitd) [inline src/components/ChatApp/MessageComposer.react.js]: Remove

**Comment by fragm3** [comment]
@akshatnitd made the changes

**Pull request merged**